bitnami/redis/templates/scripts-configmap.yaml

```diff
@@ -26,7 +26,12 @@ data:
         port_var=$(echo "${hostname^^}_SERVICE_PORT_$type" | sed "s/-/_/g")
         port=${!port_var}
 
-        if [ -z "$port" ]; then
+        {{- if and (eq .Values.architecture "replication") .Values.sentinel.enabled }}
+        if [[ -z "$port" ]] || [[ {{ .Values.sentinel.service.type | quote }} == "ClusterIP" ]]
+        {{- else }}
+        if [[ -z "$port" ]]
+        {{- end }}
+        then
```

Comment:
Why is it necessary to change the `get_port` function? 

Why if `replication` and `sentinel.enabled` and `sentinel.service.type=ClusterIP`, the `get_port` should ignore the value set in `<HOSTNAME>_SERVICE_PORT_SENTINEL` and `<HOSTNAME>_SERVICE_PORT_REDIS`?

Reply:
While I understand the behavior described here: https://github.com/bitnami/charts/issues/16946 this approach seems oddly specific and hard to understand.

If I understand correctly, this logic is required for the `useExternalDNS=true` feature, wouldn't it be better to base the conditional around that?

Reply:
In fact, external dns may not be enabled.
Let's say, I install a radis chart with a sentinel.service.type = NodePort, in the future I want to switch to an ClusterIP.  For the correct transition, I can't delete the NodePort services until complete transition to the ClusterIP (and this is where errors happens, since in pods set ENV variables like REDIS_MASTER_NODE_0_SERVICE_PORT_REDIS=30311)

Reply:
@migruiz4 what your think?